### PR TITLE
Coin app endpoints

### PR DESCRIPTION
**Pull request by haythemsellami**
HTTP resources to communicate with Coin smart contracts

### CODE REVIEW

File: lib/http/apps/coin.js [hunk 1]
```diff
@@ -31,4 +32,41 @@ app.post(`/mint`, (req, res) => {
     });
 });
 
+app.post(`/transfer`, (req, res) => {
+    let to = req.body.to;
+    let amount = req.body.amount;
+
+    res.locals.coinsence.Coin.functions.transfer(to, amount, { gasLimit: 5000000 }).then(transaction => {
```
Comment (bumi) [inline lib/http/apps/coin.js:39]: we should not add that `{ gasLimit: 5000000 }` as this is only needed locally afaik.

File: lib/http/apps/space.js [hunk 2]
```diff
@@ -32,4 +32,15 @@ app.post('/addMembers', (req, res) => {
     });
 });
 
+app.post('/leave', (req, res) => {
+
+  res.locals.coinsence.Space.functions.leaveSpace({ gasLimit: 5000000 }).then(transaction => {
```
Comment (bumi) [inline lib/http/apps/space.js]: also here, we should not add the gasLimit

**Review by bumi — CHANGES_REQUESTED** [review]

File: lib/http/apps/coin.js [hunk 3]
```diff
@@ -31,4 +32,41 @@ app.post(`/mint`, (req, res) => {
...
+        let balanceHex = result._hex;
+        let balance = result.toString();
+        res.status(201).json({ balanceHex, balance });
+    }).catch(e => {
+        console.log(e);
+        res.sendStatus(500);
+    });
+});
+
+app.get(`/supply`, (req, res) => {
```
Comment (bumi) [inline lib/http/apps/coin.js]: maybe we should use the contract names. so in this case here also totalsupply?

Comment (haythemsellami) [reply lib/http/apps/coin.js]: okay (y)

**Pull request merged**